src/Castle.Core/DynamicProxy/Generators/GeneratorUtil.cs

```diff
@@ -74,25 +76,23 @@ bool IsReadOnly(ParameterInfo parameter)
 				//
 				// The above points inform the following detection logic: First, we rely on an IL
 				// `[in]` modifier being present. This is a "fast guard" against non-`in` parameters:
-				if ((parameter.Attributes & (ParameterAttributes.In | ParameterAttributes.Out)) == ParameterAttributes.In)
-				{
-					// Here we perform the actual check. We don't rely on cmods because support
-					// for them is at current too unreliable in general, and because we wouldn't
-					// be saving much time anyway.
-					if (parameter.GetCustomAttributes(false).Any(IsIsReadOnlyAttribute))
-					{
-						return true;
-					}
-				}
+				if ((parameter.Attributes & (ParameterAttributes.In | ParameterAttributes.Out)) != ParameterAttributes.In)
+					return false;
 
-				return false;
+#if FEATURE_CUSTOMMODIFIERS
+				// Check for the required modifiers first, as it's faster.
+				// This check allows to make the detection logic more robust on the platforms which support custom modifiers.
```

Comment:
One more thing. Would it perhaps be possible to spell out *why* this check makes the detection logic "more robust"? For anyone stumbling upon this code without having followed this and my previous PR, that's perhaps not evident at all.

IIRC the reason was that DynamicProxy doesn't always reproduce the `IsReadOnlyAttribute` custom attribute placed on a parameter (and thus won't recognize them as `in` parameters); specifically, in places where the Roslyn compiler emitted a `internal` copy of it into the proxied type's assembly because it's not present in the targeted BCL. In such cases, DynamicProxy would have to (but currently doesn't) emit a copy of that attribute type into its dynamic assembly so it could apply it to the proxy method parameter.

Perhaps you could add a short explanation about this in the code comment?

Reply:
Done!